### PR TITLE
Support check health for the cluster level

### PR DESCRIPTION
**Pull request by jaime0815**
issue #19778 
Signed-off-by: yun.zhang <yun.zhang@zilliz.com>

### CODE REVIEW

**Comment by jiaoew1991** [comment]
Should we expose more detail in response about which node is healthy and which one is not,

**Comment by jaime0815** [comment]
> Should we expose more detail in response about which node is healthy and which one is not,

It should have two levels for the health check, one is the host level that has been supported with a GetComponentStates API, as you mentioned, more details about which one is healthy or not that belongs to the host level.

For the service or cluster level, the health check is used to validate whether the Milvus works well, and that more concerned with all requests can be executed successfully. this PR only checks all hosts of the cluster are healthy with  GetComponentStates,  it will be improved to use more effective checking methods in the future.

File: api/milvuspb/milvus.pb.go [hunk 1]
```diff
@@ -105,6 +105,76 @@ func (OperatePrivilegeType) EnumDescriptor() ([]byte, []int) {
...
+func (m *CheckHealthRequest) XXX_Size() int {
+	return xxx_messageInfo_CheckHealthRequest.Size(m)
+}
+func (m *CheckHealthRequest) XXX_DiscardUnknown() {
+	xxx_messageInfo_CheckHealthRequest.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_CheckHealthRequest proto.InternalMessageInfo
+
+type CheckHealthResponse struct {
```
Comment (xiaofan-luan) [inline api/milvuspb/milvus.pb.go:139]: If not healthy, should we give a reason of failed?

Comment (jaime0815) [reply api/milvuspb/milvus.pb.go:139]: We just use GetComponentStates API which is similar to Ping-Pong mode, it is a little bit difficult to give the user an available reason currently.   Maybe adding the reason is more useful after we have a more accurate usability check method?

**Pull request merged**